dashboard/src/components/DataMigrator.js

```diff
@@ -67,43 +66,51 @@ export default function useDataMigrator() {
       }
       // End of example of migration.
       */
-      // if (!organisation.migrations?.includes('reformat-observedAt-observations')) {
-      //   // some observedAt are timestamp, some are date
-      //   // it messes up the filtering by date in stats
-      //   setLoadingText(LOADING_TEXT);
-      //   const observationsRes = await API.get({
-      //     path: '/territory-observation',
-      //     query: { organisation: organisationId, after: 0, withDeleted: false },
-      //   }).then((res) => res.decryptedData || []);
+      if (!organisation.migrations?.includes('reformat-observedAt-observations-fixed')) {
+        // some observedAt are timestamp, some are date
+        // it messes up the filtering by date in stats
+        setLoadingText(LOADING_TEXT);
+        const observationsRes = await API.get({
+          path: '/territory-observation',
+          query: { organisation: organisationId, after: 0, withDeleted: false },
+        }).then((res) => res.decryptedData || []);
+
+        const observationIdsToDelete = {}; // we create an object for the loop line 87 to be fast enough
+        for (const observation of observationsRes) {
+          if (!observation.territory || !observation.team) {
+            observationIdsToDelete[observation._id] = true;
+          }
+        }
 
-      //   const newObservations = observationsRes.map((obs) => {
-      //     const observedAt = !isNaN(Number(obs.observedAt)) // i.e. is timestamp
-      //       ? dayjsInstance(Number(obs.observedAt)).toISOString()
-      //       : dayjsInstance(obs.observedAt ?? obs.createdAt).toISOString();
-      //     return {
-      //       ...obs,
-      //       user: obs.user ?? user._id, // in case of old observations missing user
-      //       observedAt,
-      //     };
-      //   });
+        const observationsWithFullData = observationsRes
+          .filter((obs) => !observationIdsToDelete[obs._id])
+          .map((obs) => {
+            const observedAt = !isNaN(Number(obs.observedAt)) // i.e. is timestamp
+              ? dayjsInstance(Number(obs.observedAt)).toISOString()
+              : dayjsInstance(obs.observedAt ?? obs.createdAt).toISOString();
```

Comment:
C'est un peu tordu et ça a des comportements inatendus potentiellement (ex: Number("") === 0, Number(null) === 0 et Number(undefined) === NaN)

Pourquoi pas juste ça vu que c'est ce que tu fais après ?

```suggestion
            const observedAt = dayjsInstance(obs.observedAt ??  obs.createdAt).toISOString()
```

Reply:
si `observedAt` est un timestamp, `dayjs(timestamp)` ne fonctionne pas...

Reply:
AH bon ?! Mais c'est nul !